arch/risc-v/src/esp32c3/esp32c3_spiflash.h

```diff
@@ -0,0 +1,251 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <nuttx/mtd/mtd.h>
+
+#ifndef __ASSEMBLY__
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32c3_spiflash_mtd
+ *
+ * Description:
+ *   Get ESP32-C3 SPI Flash MTD.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   ESP32-C3 SPI Flash MTD pointer.
+ *
+ ****************************************************************************/
+
+struct mtd_dev_s *esp32c3_spiflash_mtd(void);
+
+/****************************************************************************
+ * Name: esp32c3_spiflash_alloc_mtdpart
+ *
+ * Description:
+ *   Alloc ESP32-C3 SPI Flash MTD
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   ESP32-C3 SPI Flash MTD data pointer if success or NULL if fail
+ *
+ ****************************************************************************/
+
+FAR struct mtd_dev_s *esp32c3_spiflash_alloc_mtdpart(void);
+
+/****************************************************************************
+ * Name: esp32c3_spiflash_encrypt_mtd
+ *
+ * Description:
+ *   Get ESP32-C3 SPI Flash encryption MTD.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   ESP32-C3 SPI Flash encryption MTD pointer.
+ *
+ ****************************************************************************/
+
+struct mtd_dev_s *esp32c3_spiflash_encrypt_mtd(void);
+
+/****************************************************************************
+ * Name: spi_flash_write_encrypted
+ *
+ * Description:
+ *   Write data encrypted to Flash.
+ *
+ *   Flash encryption must be enabled for this function to work.
+ *
+ *   Flash encryption must be enabled when calling this function.
+ *   If flash encryption is disabled, the function returns
+ *   ESP_ERR_INVALID_STATE.  Use esp_flash_encryption_enabled()
+ *   function to determine if flash encryption is enabled.
+ *
+ *   Both dest_addr and size must be multiples of 16 bytes. For
+ *   absolute best performance, both dest_addr and size arguments should
+ *   be multiples of 32 bytes.
+ *
+ * Input Parameters:
+ *   dest_addr - Destination address in Flash. Must be a multiple of 16
+ *               bytes.
+ *   src       - Pointer to the source buffer.
+ *   size      - Length of data, in bytes. Must be a multiple of 16 bytes.
+ *
+ * Returned Values:
+ *   Zero (OK) is returned or a negative error.
+ *
+ ****************************************************************************/
+
+int spi_flash_write_encrypted(uint32_t dest_addr, const void *src,
```

Comment:
The following group of functions are public and thus have to be prefixed with `esp32c3_`, however I don't see them used anywhere, what do they do?

Reply:
This is a ROM function, it is provided by boards/risc-v/esp32c3/esp32c3-devkit/scripts/esp32c3_rom.ld